src/library/autodj/autodjfeature.h

```diff
@@ -62,7 +62,7 @@ class AutoDJFeature : public LibraryFeature {
     // The id of the AutoDJ playlist.
     int m_iAutoDJPlaylistId;
     AutoDJProcessor* m_pAutoDJProcessor;
-    TreeItemModel m_childModel;
+    TreeItemModel* m_pSidebarModel;
```

Comment:
For this purpose we have `parented_ptr`

Reply:
Doesn't work. Although `QAbstractItemModel` inherits from QObject, I get:

```
/home/jan/Projects/mixxx/src/util/parented_ptr.h: In instantiation of ‘parented_ptr<T>::~parented_ptr() [with T = TreeItemModel]’:
/home/jan/Projects/mixxx/src/library/itunes/itunesfeature.cpp:67:55:   required from here
/home/jan/Projects/mixxx/src/util/parented_ptr.h:31:45: error: no matching function for call to ‘TreeItemModel::parent()’
   31 |         DEBUG_ASSERT(!m_ptr || m_ptr->parent());
      |                                ~~~~~~~~~~~~~^~
```

Reply:
Caused by conflicting method names in the Qt class hierarchy and a deficiency of `parented_ptr`:

```
DEBUG_ASSERT(!m_ptr || static_cast<const QObject*>(m_ptr)->parent());
```

Please fix it and try again.